agent/xds/clusters.go

```diff
@@ -395,6 +396,30 @@ func (s *ResourceGenerator) clustersFromSnapshotMeshGateway(cfgSnap *proxycfg.Co
 		}
 	}
 
+	// Create a single cluster for local servers to be dialed by peers.
+	// When peering through gateways we load balance across the local servers. They cannot be addressed individually.
+	if cfg := cfgSnap.MeshConfig(); cfg != nil && cfg.Peering != nil && cfg.Peering.PeerThroughMeshGateways {
```

Comment:
Extracting a helper for `cfg != nil && cfg.Peering != nil && cfg.Peering.PeerThroughMeshGateways` might help with readability here too.

Reply:
Yep, done